src/vsr/replica.zig

```diff
@@ -3484,29 +3589,19 @@ pub fn ReplicaType(
         fn ignore_view_change_message(self: *const Self, message: *const Message) bool {
             assert(message.header.command == .do_view_change or
                 message.header.command == .start_view);
-            assert(message.header.view > 0); // The initial view is already zero.
             assert(self.status != .recovering); // Single node clusters don't have view changes.
             assert(message.header.replica < self.replica_count);
 
             const command: []const u8 = @tagName(message.header.command);
 
-            if (self.status == .recovering_head and message.header.command != .start_view) {
-                log.debug("{}: on_{s}: ignoring (recovering_head)", .{ self.replica, command });
-                return true;
-            }
-
             if (message.header.view < self.view) {
                 log.debug("{}: on_{s}: ignoring (older view)", .{ self.replica, command });
                 return true;
             }
 
-            if (message.header.view == self.view and self.status == .normal) {
-                log.debug("{}: on_{s}: ignoring (view started)", .{ self.replica, command });
-                return true;
-            }
```

Comment:
This now only impacts DVCs (moved into the switch) — status=normal is now permitted to handle same-view SVs.